set-web/src/main/java/eu/europeana/set/web/utils/UserSetSearchApiUtils.java

```diff
@@ -0,0 +1,142 @@
+package eu.europeana.set.web.utils;
+
+import eu.europeana.api.commons.definitions.vocabulary.CommonApiConstants;
+import eu.europeana.api.commons.web.exception.HttpException;
+import eu.europeana.set.definitions.model.UserSet;
+import eu.europeana.set.definitions.model.vocabulary.WebUserSetFields;
+import eu.europeana.set.search.SearchApiRequest;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+public class UserSetSearchApiUtils {
+
+    /**
+     * Will create the Serach Api post request url
+     * eg : https://api.europeana.eu/record/v2/search.json?wskey=api2demo
+     *
+     * @param userSet
+     * @param apiKey
+     * @return
+     */
+    public String buildSearchApiPostUrl(UserSet userSet, String apiKey, String searchUrl) {
+        StringBuilder url = new StringBuilder();
+        if (!userSet.isOpenSet()) {
+            url.append(getBaseSearchUrl(searchUrl));
+        } else {
+            url.append(StringUtils.substringBefore(userSet.getIsDefinedBy(), "?"));
+        }
+        // add apikey
+        url.append('?').append(CommonApiConstants.PARAM_WSKEY).append('=').append(apiKey);
+        return url.toString();
+    }
+
+    /**
+     * Returns the Search APi post Request body
+     * @param userSet
+     * @param pageSize
+     * @return
+     */
+    public SearchApiRequest buildSearchApiPostBody(UserSet userSet, String sort, String sortOrder, int pageNr, int pageSize) {
+        if (!userSet.isOpenSet()) {
+            return buildSearchApiPostBodyForClosedSets(userSet, pageSize);
+        }
+        SearchApiRequest searchApiRequest = new SearchApiRequest();
+        // remove pagination and ordering
+        Integer start = pageNr * pageSize + 1;
+
+        searchApiRequest.setQuery(getQueryParamFromURL(userSet.getIsDefinedBy()));
+        searchApiRequest.setStart(start);
+        searchApiRequest.setRows(pageSize);
+
+        if(sort != null && sortOrder == null) {
+            searchApiRequest.setSort(new String[]{sort});
+        }
+        if (sort != null && sortOrder != null) {
+            searchApiRequest.setSort(new String[]{sort + " " + sortOrder});
+        }
+        return searchApiRequest;
+    }
+
+    /**
+     * This method retrieves item ids from the closed userSet to build SearchApiRequest.
+     * e.g.
+     * {query='europeana_id:("/165/https___bibdigital_rjb_csic_es_idviewer_11929_40" OR "/2020903/KKSgb2947_97")',
+     * start=1, rows=5, sort=europeana_id desc}
+     *
+     * @param userSet
+     * @param pageSize
+     * @return
+     * @throws HttpException
+     */
+    SearchApiRequest buildSearchApiPostBodyForClosedSets(UserSet userSet, int pageSize) {
```

Comment:
shouldn't this method have the "int page" parameter as well?

Reply:
added pageNr